clients/runtime/src/types.rs

```diff
@@ -114,6 +115,34 @@ mod metadata_aliases {
 	}
 }
 
+pub mod currency_id {
+	use primitives::Asset;
+
+	use super::*;
+	use crate::Error;
+
+	pub trait CurrencyIdExt {
+		fn inner(&self) -> Result<String, Error>;
+	}
+
+	impl CurrencyIdExt for CurrencyId {
+		fn inner(&self) -> Result<String, Error> {
+			match self {
+				CurrencyId::Native => Ok("Native".to_owned()),
+				CurrencyId::XCM(foreign_currency_id) =>
+					Ok(format!("XCM {}", foreign_currency_id).to_owned()),
+				CurrencyId::Stellar(stellar_asset) => match stellar_asset {
+					Asset::StellarNative => Ok("Stellar Native".to_owned()),
+					Asset::AlphaNum4 { code, issuer } =>
+						Ok(format!("Stellar asset <{:?}>:<{:?}>", code, issuer)),
+					Asset::AlphaNum12 { code, issuer } =>
+						Ok(format!("Stellar asset <{:?}>:<{:?}>", code, issuer)),
```

Comment:
Done.

Reply:
Thanks for changing it. But I checked and saw that this now of course shows the actual bytes and no nice human-readable asset code and issuer. Can you change it so that code and issuer are shown as an ascii string instead of the bytes?

Reply:
Done.